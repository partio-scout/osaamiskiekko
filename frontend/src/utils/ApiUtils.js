export const addTypeToSchoolOrOrganization = (schoolOrOrganization, type) => schoolOrOrganization.map(item => {
    if (type === 'school') {
      const o = Object.assign({}, item);
      o.type_fi = 'Oppilaitos';
      o.type_en = 'School';
      o.type_se = 'Skola'
      return o;
    }
    return null;
  })