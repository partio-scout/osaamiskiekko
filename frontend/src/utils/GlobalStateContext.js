import React from 'react';
// import { defaultTheme } from '../styles/Themes';

export const useGlobalStateContext = () => React.useContext(GlobalStateContext);

const defaultValues = {
  language: 'fi',
  changeLanguage: (language) => {  console.log(language) },
  theme: 'defaultTheme',
  changeTheme: (theme) => { console.log(theme) },
};

const GlobalStateContext = React.createContext(defaultValues);

export default GlobalStateContext;
