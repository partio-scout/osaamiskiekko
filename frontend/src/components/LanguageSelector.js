import React from 'react';
import styled from 'styled-components';
import { useGlobalStateContext } from '../utils/GlobalStateContext';

const S = {};
S.LanguageSelector = styled.div`
position: relative;

select {
  height: 49px;	
  width: 49px;	
  border-radius: 24.5px;	
  background-color: #FFFFFF;
  color: ${props => props.theme.colors.osaamisKiekkoBlue};
  padding: 5px 5px 5px 10px;
  border: none;
  font-size: 14px;
  -webkit-appearance: button;
  appearance: button;
  outline: none;
}

::before {
  content: "\f0d7";
  font-family: FontAwesome;
  position: absolute;
  font-size: 5px;
  top: 0;
  right: 10px;
  width: 20%;
  height: 100%;
  text-align: center;
  font-size: 15px;
  line-height: 50px;
  color: rgba(0, 0, 0, 0.5);
  background-color: rgba(255, 255, 255, 0.1);
  pointer-events: none;
  :hover {
    color: rgba(0, 0, 0, 0.6);
    background-color: rgba(255, 255, 255, 0.2);
  }
}
`;

export default function LanguageSelector() {
  const { changeLanguage } = useGlobalStateContext();
  const setLanguage = (e) => changeLanguage(e.target.value);

  return (
    <S.LanguageSelector>
      <select onChange={setLanguage}>
        <option value="fi">Fi</option>
        <option value="en">En</option>
        <option value="sv">Sv</option>
      </select>
    </S.LanguageSelector>
  )
}
