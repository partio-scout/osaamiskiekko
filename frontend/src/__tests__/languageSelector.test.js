import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderWithProviders from '../testHelpers'
import LanguageSelector from '../components/LanguageSelector';
import * as GlobalStateContext from '../utils/GlobalStateContext';

configure({adapter: new Adapter()});

describe('language selector', () => {
  test('snapshot matches', () => {
    const wrapper = shallow(
      <LanguageSelector />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('change affects globalState', () => {
    const mockChangeLanguage = jest.fn(lang => { globalValues.language = lang });
    const globalValues = {
      language: 'fi',
      changeLanguage: mockChangeLanguage
    };
    jest.spyOn(GlobalStateContext, 'useGlobalStateContext')
      .mockImplementation(() => globalValues);

    const wrapper = shallow(
      <LanguageSelector />
    );

    wrapper.find('select').simulate('change', {target: { value : 'en'}});
    wrapper.find('select').simulate('change', {target: { value : 'sv'}});

    expect(mockChangeLanguage.mock.calls.length).toBe(2);
    expect(globalValues.language).toBe('sv');
  });
});
