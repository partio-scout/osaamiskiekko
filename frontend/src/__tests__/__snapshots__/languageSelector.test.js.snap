// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language selector snapshot matches 1`] = `
<styled.div>
  <select
    onChange={[Function]}
  >
    <option
      value="fi"
    >
      Fi
    </option>
    <option
      value="en"
    >
      En
    </option>
    <option
      value="sv"
    >
      Sv
    </option>
  </select>
</styled.div>
`;
